### PR TITLE
Platform page & preset page

### PR DESCRIPTION
**Pull request by ethteck**
This PR adds a platform page (/platform/n64) and preset page (/preset/32) to the site, allowing users to view scratches for a given platform or preset. Eventually, we can add more to these pages.

I linked to these from the About panel, but there are probably other places we can add links. I didn't look too hard

### CODE REVIEW

**Comment by ethteck** [comment]
gonna add a couple things today or tomorrow, so marking as draft temporarily

File: frontend/src/app/(navfooter)/preset/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        <div className="flex items-center gap-2 text-2xl font-medium">
+            <PlatformIcon platform={preset.platform} size={32} />
+            <h1>
+                {preset.name}
+            </h1>
+        </div>
+        <p className="py-3 text-gray-11">{compilerName} • {preset.num_scratches} scratches</p>
+
+        <section>
+            <h2 className="pb-2 text-lg font-medium tracking-tight">Scratches</h2>
```
Comment (nanaian) [inline frontend/src/app/(navfooter)/preset/[id]/page.tsx:53]:
```suggestion
        <p className="py-3 text-gray-11">{compilerName}</p>

        <section>
            <h2 className="pb-2 text-lg font-medium tracking-tight">Scratches ({preset.num_scratches})</h2>
```

File: frontend/src/app/(navfooter)/platform/[id]/page.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        <div className="flex items-center gap-2 text-2xl font-medium">
+            <PlatformIcon platform={platform.id} size={32} />
+            <h1>
+                {platform.name}
+            </h1>
+        </div>
+        <p className="py-3 text-gray-11">{platform.description} • {platform.num_scratches} scratches</p>
+
+        <section>
+            <h2 className="pb-2 text-lg font-medium tracking-tight">Scratches</h2>
```
Comment (nanaian) [inline frontend/src/app/(navfooter)/platform/[id]/page.tsx:48]:
```suggestion
        <p className="py-3 text-gray-11">{platform.description}</p>

        <section>
            <h2 className="pb-2 text-lg font-medium tracking-tight">Scratches ({platform.num_scratches})</h2>
```

**Review by nanaian — COMMENTED** [review]
I'd suggest moving the scratch count to inside the heading to colocate information but nbd

**Pull request merged**